lib/_tls_legacy.js

```diff
@@ -585,14 +585,12 @@ function onhandshakestart() {
   ssl.lastHandshakeTime = now;
   if (first) return;
 
-  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT) {
+  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT && self.cleartext) {
     // Defer the error event to the next tick. We're being called from OpenSSL's
     // state machine and OpenSSL is not re-entrant. We cannot allow the user's
     // callback to destroy the connection right now, it would crash and burn.
-    setImmediate(function() {
-      var err = new Error('TLS session renegotiation attack detected');
-      if (self.cleartext) self.cleartext.emit('error', err);
-    });
+    const err = new Error('TLS session renegotiation attack detected');
+    process.nextTick(() => self.cleartext.emit('error', err));
```

Comment:
Does this really need to be changed since `setImmediate()` will already delay it?